delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Projects.scala

```diff
@@ -68,18 +68,19 @@ trait Projects {
   def fetch(ref: ProjectRef): UIO[Option[ProjectResource]]
 
   /**
-    * Fetches the current active project, rejecting if the project does not exists or if the project is deprecated
+    * Fetches and validate the project, rejecting if the project does not exists or if the project/its organization is deprecated
+    * @param ref                   the project reference
+    * @param rejectionMapper  allows to transform the ProjectRejection to a rejection fit for the caller
     */
-  def fetchActiveProject(ref: ProjectRef): IO[ProjectRejection, Project]
+  def fetchActiveProject[R](ref: ProjectRef)(implicit rejectionMapper: Mapper[ProjectRejection, R]): IO[R, Project]
 
   /**
-    * Fetches the current project, rejecting if the project does not exists
+    * Fetches the current project from cache, rejecting if the project does not exists
+    *
+    * @param ref the project reference
+    * @param rejectionMapper  allows to transform the ProjectRejection to a rejection fit for the caller
     */
-  def fetchProject(ref: ProjectRef): IO[ProjectRejection, Project] =
-    fetch(ref).flatMap {
-      case Some(resource) => IO.pure(resource.value)
-      case None           => IO.raiseError(ProjectNotFound(ref))
-    }
+  def fetchFromCache[R](ref: ProjectRef)(implicit rejectionMapper: Mapper[ProjectRejection, R]): IO[R, Project]
```

Comment:
This is misleading. The cache should be used only for listings. We would rather use the state actor, which is going to be alive all the time anyway, given our configuration. So I would skip that to avoid confusion